src/doxygen.cpp

```diff
@@ -2732,7 +2732,14 @@ static MemberDef *addVariableToFile(
   //md->setOuterScope(fd);
   if (!root->explicitExternal)
   {
-    mmd->setBodySegment(root->startLine,root->bodyLine,root->endBodyLine);
+    if (mmd->isEnumValue())
+    {
+      mmd->setBodySegment(root->bodyLine,root->bodyLine,root->endBodyLine);
```

Comment:
I think it is better to make sure `startLine` is set correctly also for enum values, rather than making an exception here.

Reply:
I do agree, but was (initially) not able to find a better solution.

Did some further tests and also the following seems to work:
```
diff --git a/src/scanner.l b/src/scanner.l
index 35850404e6..2fcdf96084 100644
--- a/src/scanner.l
+++ b/src/scanner.l
@@ -4060,8 +4060,16 @@ NONLopt [^\n]*
                                           if (!yyextra->current->name.isEmpty())
                                           {
                                             yyextra->current->fileName   = yyextra->fileName;
-                                            yyextra->current->startLine  = yyextra->yyLineNr;
-                                            yyextra->current->startColumn = yyextra->yyColNr;
+                                            if (yyextra->current_root->section.isEnum() || yyextra->current_root->spec.isEnum())
+                                            {
+                                              yyextra->current->startLine  = yyextra->current->bodyLine;
+                                              yyextra->current->startColumn = yyextra->current->bodyColumn;
+                                            }
+                                            else
+                                            {
+                                              yyextra->current->startLine  = yyextra->yyLineNr;
+                                              yyextra->current->startColumn = yyextra->yyColNr;
+                                            }
                                             if (!yyextra->current_root->spec.isEnum())
                                             {
                                               yyextra->current->type       = "@"; // enum marker
```

though this relies on the fact that somewhere a `,` is added at the end of the text for a second pass (`parseCompound`) of the scanner, though I don't see where this `,` is added. This makes me a bit reluctant to create a new PR for it.

Reply:
Found the adding of `,`:
```
yyextra->current->program << ','; // add field terminator
```
this is done twice for enum values.

I've just pushed a new proposed patch, pull request #11145 that can supersede this one.